doc/http.rst

```diff
@@ -0,0 +1,151 @@
+Commissaire HTTP
+================
+
+Commissaire HTTP provides a multithreaded REST interface into Commissaire
+functionality. The server is broken up into 5 main parts: ``Router``,
+``Dispatcher``, ``handlers``, and the ``CommissaireHttpServer`` itself.
+
+Router
+------
+The ``Router`` maps URI paths to controllers. The following example would route
+the path /hello/ to the controller at ``commissaire_http.handlers.hello_world``
+if the HTTP method is ``GET``.
+
+.. code-block:: python
+
+   mapper = Router()
+   mapper.connect(
+       '/hello/',
+       controller='commissaire_http.handlers.hello_world',
+       conditions={'method': 'GET'})
+
+
+Dispatcher
+----------
+The ``Dispatcher`` uses a ``Router`` and, as it's name suggests, dispatches
+the requests to the proper controller. It also takes care of loading
+handlers. The following example creates a new ``Dispatcher`` instance using
+a previously created ``Mapper``. It will load handlers found in the
+``commissaire_http.handlers`` and ``mypackage.handlers`` packages.
+
+.. code-block:: python
+
+   dispatcher = Dispatcher(
+       mapper,
+       handler_packages=[
+        'commissaire_http.handlers',
+        'mypackage.handlers'])
+
+
+Handlers
+--------
+``Handlers``, also called controllers, do the majority of the business logic.
+A ``Handler`` can be a function or a class, but must follow a specific
+convention so the ``Dispatcher`` knows it's valid during loading.
+
+Function Handler
+````````````````
+Function handlers must take two parameters: ``message`` and ``bus``. The first
+input, ``message``, is the jsonrpc message. The second input, ``bus`` will
+either be a valid connection to the bus or, if the bus is not enabled,
+``None``.
+
+When referencing a function handler as a controller use the full package path
+to the function If the function is ``hello_world`` and it lives under
+``commissaire_http.handlers`` then the controller would be
+``commissaire_http.handlers.hello_world``.
+
+The following example would show the user ``{"Hello": "there"}`` or
+``{"Hello", "{{ name }}"}`` depending parameters. Remember, the return of
+the handler must be a valid jsonrpc message as well!
+
+.. note::
+
+    The ``method`` in the incoming jsonrpc message is hikjacked and filled
```

Comment:
typo: "hijacked" not "hikjacked" ... although yours sounds cooler  :stuck_out_tongue_closed_eyes: